### PR TITLE
[KernelGen][Nvidia] Add special_psi operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `special_psi`.

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (15 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Performance
Test command: `pytest benchmark/test_special_psi.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 139.583 | 16.058 | 8.692 |
| 64, 64 | 0.023 | 0.007 | 3.406 |
| 4096, 4096 | 2.202 | 0.342 | 6.439 |
| 64, 512, 512 | 2.202 | 0.342 | 6.446 |
| 1024, 1024, 1024 | 139.583 | 16.059 | 8.692 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 139.628 | 16.230 | 8.603 |
| 64, 64 | 0.023 | 0.007 | 3.166 |
| 4096, 4096 | 2.203 | 0.343 | 6.429 |
| 64, 512, 512 | 2.202 | 0.342 | 6.443 |
| 1024, 1024, 1024 | 139.621 | 16.230 | 8.603 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 139.593 | 16.233 | 8.599 |
| 64, 64 | 0.023 | 0.007 | 3.156 |
| 4096, 4096 | 2.202 | 0.340 | 6.484 |
| 64, 512, 512 | 2.202 | 0.340 | 6.481 |
| 1024, 1024, 1024 | 139.585 | 16.233 | 8.599 |

**Arithmetic Mean Speedup: 6.683x**

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 6.683 | Primary |
| Tianshu | PASS | PASS (42 cases) | 1.69 | — |
| Muxi | PASS | PASS (42 cases) | 1.00 | — |
| Ascend | PASS | — | — | — |
| Hygon | PASS | PASS (42 cases) | 1.26 | — |

## Files Changed
- `src/flag_gems/ops/special_psi.py`: Triton kernel implementation
- `tests/test_special_psi.py`: Accuracy test
- `benchmark/test_special_psi.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.4)

### CODE REVIEW

File: src/flag_gems/ops/special_psi.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import triton.language as tl
+
+from flag_gems.utils import pointwise_dynamic
+
+logger = logging.getLogger(__name__)
+
+
+@pointwise_dynamic(promotion_methods=[(0, "INT_TO_FLOAT")])
+@triton.jit
+def digamma_kernel(x):
```
Comment (Caeruleann) [inline src/flag_gems/ops/special_psi.py]:
```suggestion
def special_psi_kernel(x):
```

File: tests/test_special_psi.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import flag_gems
+
+from . import accuracy_utils as utils
+
+
+@pytest.mark.special_psi
+@pytest.mark.parametrize("shape", utils.POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_special_psi(shape, dtype):
+    inp = torch.rand(shape, dtype=dtype, device=flag_gems.device) + 1.0
```
Comment (Caeruleann) [inline tests/test_special_psi.py]: for Add test cases for negative numbers.

**Comment by Caeruleann** [comment]
`inp = torch.rand(shape, dtype=dtype, device=flag_gems.device) + 1.0`
`torch.special.psi()` supports negative numbers; do not use `rand() + 1.0` to generate test data.

File: tests/test_special_psi.py (hunk 2 above)
Comment (bwbwzzz) [reply tests/test_special_psi.py]: have added

File: tests/test_special_psi.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@pytest.mark.parametrize("shape", utils.POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_special_psi(shape, dtype):
+    # psi (digamma) is defined on both positive and negative reals, with
+    # poles at 0 and the negative integers. Cover both signs while keeping
+    # inputs away from the poles: an integer base in [-4, 4) plus a
+    # fractional part in [0.2, 0.8] so values never land on an integer.
+    base = torch.randint(-4, 4, shape, device=flag_gems.device).to(dtype)
+    frac = torch.rand(shape, dtype=dtype, device=flag_gems.device) * 0.6 + 0.2
+    inp = base + frac
```
Comment (Caeruleann) [inline tests/test_special_psi.py]:
```suggestion
    inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
```

Comment (Caeruleann) [reply tests/test_special_psi.py]: The `torch. special_psi` operator can handle normal floating-point numbers, including negative values. Use the `torch. randn` interface to generate inp data.

Comment (bwbwzzz) [reply tests/test_special_psi.py]: thanks， have apply suggestion

**Comment by bin913** [comment]
@bwbwzzz Please fix failing check
